journal/202104.md

```diff
@@ -0,0 +1,351 @@
+- allow to specify interfaces to listen by [names](https://github.com/decred/dcrd/pull/2623) in addition to exact IP addresses
+
+<a id="dcrwallet" />
+
+**[dcrwallet](https://github.com/decred/dcrwallet)**
+
+- a new method that returns wallet's [sync status](https://github.com/decred/dcrwallet/pull/1991)
+- handle ticket status when the VSP has [received](https://github.com/decred/dcrwallet/pull/1965) its fee tx but has not yet broadcast it
+- fixed bugs found when used by DCRDEX
+
+<a id="decrediton" />
+
+**[Decrediton](https://github.com/decred/decrediton)**
+
+- initial [DEX integration](https://github.com/decred/decrediton/pull/3356) { how does it work, i.e. does it run a one more daemon? is there a BTC wallet? UI shots? }
+- a form for [funding](https://github.com/decred/decrediton/pull/3426) the DEX account from the registration view
+- allow to use an [existing](https://github.com/decred/decrediton/pull/3438) wallet account for the DEX instead of making a new one
+- added [QR code](https://github.com/decred/decrediton/pull/3112) generation to export active tickets (to be used with the Decred Address Scanner app)
+- [remember](https://github.com/decred/decrediton/pull/3441) ticket auto-buyer settings
+- added a migration to initialize [account](https://github.com/decred/decrediton/pull/2664) passphrases with the wallet passphrase. Also, some operations (send, purchase ticket, mixing, revoke) switched to only unlock a single needed account instead of unlocking an entire wallet.
+- lock [new accounts](https://github.com/decred/decrediton/pull/3424) after creation
+- allow restoring hex seeds up to 128 characters (including BIP0039 seeds)
+- updated design of [Transactions](https://github.com/decred/decrediton/pull/3326), [Proposals](https://github.com/decred/decrediton/pull/3345) and [Purchase tickets](https://github.com/decred/decrediton/pull/3349) views
+- many smaller design tweaks
+- switched to a strict Protobuf generation without `eval()` and restricted to only access exteral servers via `https` in production builds using the [CSP header](https://github.com/decred/decrediton/pull/3366)
+- refactoring of state management, separation of concerns, dependency upgrades
+- expanded test coverage
+- ~24 bug fixes
+
+Release [v1.6.3](https://github.com/decred/decrediton/tree/release-v1.6%2B) is being prepared.
+
+<a id="politeia" />
+
+**[Politeia](https://github.com/decred/politeia)**
+
+- list [proposals](https://github.com/decred/politeiagui/pull/2340) from [proposals-archive.decred.org](https://proposals-archive.decred.org/) on the new version
+- [verify](https://github.com/decred/politeia/pull/1377) comment and vote bundles in the `politeiaverify` tool
+- add Unix [timestamp](https://github.com/decred/politeia/pull/1395) to cast votes { is there no temporal batching due to tstore now? does it lower privacy? }
```

Comment:
The timestamp was added to the vote structure so that dcrdata can build their vote graphs ([example](https://dcrdata.decred.org/proposal/video-content-production-for-decred-phase-3)) using only the politeiawww API.

This was not possible in the git backend because of how it was setup.  Adding a commit in git is an expensive operation.  The cast votes were only added to the public git repo once per hour in order to reduce the number of commits being made.  This means that you could only determine if a vote was cast within a +/- 60 minute period. Adding the timestamp to the vote would have provided granular details that could be used to correlate blocks of votes to their owner.  Since there was no timestamp for dcrdata to key off of, dcrdata would pull in the actual git repos and parse the votes manually in order to create their vote graphs.

When you cast a vote in the tstore backend the timestamp is already being recorded by the trillian instance.  There's no way to prevent this.  The timestamp was added to the vote structure because the trillian timestamp is already public data and adding it directly to the vote structure allows dcrdata to build their vote graphs using only data pulled from the politeiawww API instead of having to walk the trillian trees and check the timestamps manually.

The tstore backend does lower privacy for people casting all of their votes at once, but there is not much we can do about this without writing our own trillian implementation.  People who want to protect their privacy MUST use the `politeiavoter` trickling feature.

Reply:
Thanks for the detailed clarification. I added a compressed version as a separate paragraph in 1c37ddc